src/Orleans/CodeGeneration/TypeUtils.cs

```diff
@@ -561,7 +561,7 @@ public static IEnumerable<TypeInfo> GetDefinedTypes(Assembly assembly, Logger lo
             }
             catch (Exception exception)
```

Comment:
Why not just catching `ReflectionTypeLoadException` here and make others bubble up? This way we're swallowing all other exceptions if logging is not turned on to at least Warning level.